Libraries/LibWeb/CSS/SelectorEngine.cpp

```diff
@@ -867,6 +867,11 @@ static bool fast_matches_simple_selector(CSS::Selector::SimpleSelector const& si
 static bool fast_matches_compound_selector(CSS::Selector::CompoundSelector const& compound_selector, Optional<CSS::CSSStyleSheet const&> style_sheet_for_rule, DOM::Element const& element, GC::Ptr<DOM::Element const> shadow_host)
```

Comment:
This commit only touches the non-recursive `fast_matches` code path. What about the recursive `matches` path?

See `can_use_fast_matches` for how we decide whether a selector is fast-matchable :)

Also, please add a test!